### PR TITLE
fix: specify nonroot uid for manager

### PR DESCRIPTION
**Pull request by heww**
See https://github.com/kubernetes-sigs/kubebuilder/pull/1635 for more
info

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,6 @@
 FROM gcr.io/distroless/static:nonroot
 WORKDIR /
 COPY manager .
-USER nonroot:nonroot
+USER 65532:65532
```
Comment (bitsf) [inline Dockerfile:6]: 这个数是随便写的吗？咱们有没有统一的规范

Comment (heww) [reply Dockerfile:6]: This user id is from https://github.com/kubernetes-sigs/kubebuilder/pull/1635 which is the part of the kubebuilder v3.

Comment (bitsf) [reply Dockerfile:6]: 但是我看harbor这边用的都是10000
是不是我们应该统一

Comment (heww) [reply Dockerfile:6]: This is the user id to run the harbor-operator itself, not the user id to run the harbor. 

I think we should keep it the same with kubebuilder because these files may be generated by the kubebuilder.

File: charts/harbor-operator/README.md [hunk 2]
```diff
@@ -30,7 +30,6 @@ Deploy Harbor Operator
 | nodeSelector | object | `{}` | Expects input structure as per specification <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#nodeselector-v1-core> For example: `[   {     "matchExpressions": [       {         "key": "kubernetes.io/e2e-az-name",         "operator": "In",         "values": [           "e2e-az1",           "e2e-az2"         ]       }     ]   } ]` |
 | podAnnotations | object | `{}` | Additional annotations to add to the controller Pods |
 | podLabels | object | `{}` | Additional labels to add to the controller Pods |
-| podSecurityContext | object | `{}` | Expects input structure as per specification <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podsecuritycontext-v1-core> For example: `{   "fsGroup": 2000,   "runAsUser": 1000,   "runAsNonRoot": true }` |
```
Comment (steven-zou) [inline charts/harbor-operator/README.md:33]: Why did you remove this section?

Comment (heww) [reply charts/harbor-operator/README.md:33]: Because the operator manager default runs as a non-root user and runs as a user 65532, the `runAsUser` and `runAsNonRoot` is not needed now.

If the podSecurityContext not be removed, the customer needs to provide the podSecurityContext to run the operator in the restricted psp env.

Comment (steven-zou) [reply charts/harbor-operator/README.md:33]: If the users want to override the run user, they can also provide this section `podSecurityContext`, right? @heww

**Review by bitsf — APPROVED** [review]

File: charts/harbor-operator/README.md (hunk 2 above)
Comment (bitsf) [reply charts/harbor-operator/README.md:33]: 他已经在dockerfile中写死user id了，这个参数估计没有了

Comment (bitsf) [reply charts/harbor-operator/README.md:33]: 他已经在dockerfile中写死user id了，这个参数估计没有了

**Review by steven-zou — APPROVED** [review]
LGTM

File: charts/harbor-operator/README.md (hunk 2 above)
Comment (heww) [reply charts/harbor-operator/README.md:33]: I tested that the pod also runs successes when providing another user id in the security context.

File: Dockerfile (hunk 1 above)
Comment (holyhope) [reply Dockerfile:6]: I cannot read what @bitsf wrote,
But I agree with @heww : we should keep the `nonroot` user:
https://github.com/GoogleContainerTools/distroless/blob/master/examples/nonroot/BUILD#L17

Why using it here?
IMO The change should only be be applied to files under `config/`.

Comment (heww) [reply Dockerfile:6]: This PR lets the harbor operator can run in the k8s cluster with PSP enabled.

The `config/manager/manager.yaml` was changed with `runAsUser: 65532`

Comment (holyhope) [reply Dockerfile:6]: I got it, but the dockerfile does not require any changes.
The user id should be changed on runtime, not at buildtime.

The user `65532` does not exist in `gcr.io/distroless/static:nonroot` image.
